File: lib/domain/usecases/task/delete_task_usecase.dart
import '../../entities/commit.dart' as entity;
import '../../entities/enums.dart';
import '../../repositories/i_commit_repository.dart';
import '../../repositories/i_task_repository.dart';

/// 删除任务用例
class DeleteTaskUseCase {
  DeleteTaskUseCase(
    this._taskRepository,
    this._commitRepository,
  );

  final ITaskRepository _taskRepository;
  final ICommitRepository _commitRepository;

  Future<void> execute(String taskId) async {
    final task = await _taskRepository.getById(taskId);
    if (task == null) {
      throw ArgumentError('任务不存在: $taskId');
    }

    // 记录删除提交
    await _commitRepository.create(
      entity.Commit.create(
        taskId: task.id,
        branchId: task.branchId,
        message: '删除任务: ${task.title}',
        type: CommitType.delete,
      ),
    );

    await _taskRepository.delete(taskId);
  }
}

File: web/src/application/usecases/task/delete-task-usecase.ts
import { Commit } from '../../../domain/entities/commit';
import { CommitType } from '../../../domain/entities/enums';
import { ICommitRepository } from '../../../domain/repositories/i-commit-repository';
import { ITaskRepository } from '../../../domain/repositories/i-task-repository';
import { validateNotEmpty } from '../../../core/utils/validators';

export class DeleteTaskUseCase {
  constructor(
    private taskRepo: ITaskRepository,
    private commitRepo: ICommitRepository
  ) {}

  async execute(id: string): Promise<void> {
    validateNotEmpty(id, '任务 ID');
    const task = await this.taskRepo.getById(id);
    if (!task) {
      throw new Error('任务不存在');
    }
    await this.taskRepo.delete(id);
    const commit = Commit.create(
      id,
      task.branchId,
      `删除任务 ${task.title}`,
      CommitType.delete
    );
    await this.commitRepo.create(commit);
  }
}
